taxjar/taxjar-magento2-extension
# INT-1892: Backup rates progress indicator

### Context

The existing backup rates indicator shows how many rates exist, not specifically TaxJar backup rates. It is not a true indicator of how many of the available backup rates have been imported.

### Description

- Adds global configuration value for expected TaxJar backup rates count.
- Updates Backup Rates UI template (`backup.phtml`) and "controller" `Block.php` to provide data values from Block.php and render HTML in template (instead of concatenating a raw HTML string in Block.php)
- Removes Backup Rates UI element(s) for displaying individual states and their tax rate counts.
- Replaces "magic" strings with framework const values where applicable
- Updates cache flush method in `ImportRates.php`

![Screen Shot 2021-08-24 at 11 13 03 AM](https://user-images.githubusercontent.com/47947793/130746466-ee06002e-1b19-446f-a848-0ae2e182f203.png)

![Screen Shot 2021-08-24 at 11 49 27 AM](https://user-images.githubusercontent.com/47947793/130746495-1f0dda76-7f77-4f00-a14b-ffa33defddab.png)

### Performance

There is a somewhat negligible performance ding due to the cache clearing that is required to accurately display expected backup rates count when value is updated in global config.

### Testing

1. With TJ extension enabled and nexus synced in Magento 2, navigate to `Stores > Configuration > Sales > Tax`
2. Enable backup rates, then select "Save Config"
3. Observe values set for expected rates count and last updated at date.
4. Wait for async rates to process and observe that actual rates count matches expected rates count. (If verifying against DB values, note that any values that are not associated with TaxJar's `tax_calculation_rules` entry are not included in rate count, and the expected rates count value is set on table `core_config_data`.
5. Disable backup rates, then select "Save Config"
6. Wait for async rates delete process to finish, then observe TaxJar tax rule and related rates are deleted. Backup rates count value is set to `0` on table `core_config_data`.

#### Versions
